File: runtime-testsuite/test/org/antlr/v5/test/runtime/java/helpers/CustomStreamErrorListener.java
package org.antlr.v5.test.runtime.java.helpers;

import org.antlr.v5.runtime.java.BaseErrorListener;
import org.antlr.v5.runtime.core.error.RecognitionException;
import org.antlr.v5.runtime.core.Recognizer;

import java.io.PrintStream;

public class CustomStreamErrorListener extends BaseErrorListener {
	private final PrintStream printStream;

	public CustomStreamErrorListener(PrintStream printStream){
		this.printStream = printStream;
	}

	@Override
	public void syntaxError(Recognizer<?, ?> recognizer,
							Object offendingSymbol,
							int line,
							int charPositionInLine,
							String msg,
							RecognitionException e) {
		printStream.println("line " + line + ":" + charPositionInLine + " " + msg);
	}
}

File: runtime-testsuite/test/org/antlr/v5/test/runtime/kotlin/helpers/CustomStreamErrorListener.kt
package org.antlr.v5.test.runtime.kotlin.helpers

import org.antlr.v5.runtime.kotlin.BaseErrorListener
import org.antlr.v5.runtime.core.error.RecognitionException
import org.antlr.v5.runtime.core.Recognizer
import java.io.PrintStream

class CustomStreamErrorListener(private val printStream: PrintStream) : BaseErrorListener() {
    override fun syntaxError(
        recognizer: Recognizer<*, *>,
        offendingSymbol: Any?,
        line: Int,
        charPositionInLine: Int,
        msg: String,
        e: RecognitionException?
    ) {
        printStream.println("line $line:$charPositionInLine $msg")
    }
}
